### ISSUE TITLE
Recurring Events

### ISSUE THREAD

**Issue by flootr** [peerigon/scrapegoat#12]
There is a problem with recurring events when fetching events by time. If an even takes place on monday and recurs weekly it is not fetched after the original event ends. Need to fix this!

**Comment by flootr** [comment]
Fail. Scrapegoat is not buggy.. it's me.

**Comment by flootr** [comment]
Seems to be the problem is something else. Scrapegoat fetches the events correctly, but the start/end properties are the same as the ones on the original event. Do not know if this is expected behaviour, but I have to make this clear - otherwise it is really confusing.

**Comment by flootr** [comment]
Note for me:
A recurring event looks like the following:
```
BEGIN:VEVENT
[...]
RRULE:FREQ=DAILY;INTERVAL=1;COUNT=3
END:VEVENT
```
The `RRULE` attribute tells us something about the frequency, interval and how often the event is actually recurring. - **NOTE:** this is the state, if none of the 'child-events' have been edited (e.g. in an app or so).

If I edit a child event, do another request, our response looks like this:
```
BEGIN:VEVENT
[...]
UID:AE1882FC-E25C-43E7-B398-CAAAB9242AF0
RRULE:FREQ=DAILY;INTERVAL=1;COUNT=3
END:VEVENT

BEGIN:VEVENT
[...]
UID:AE1882FC-E25C-43E7-B398-CAAAB9242AF0
RECURRENCE-ID;TZID=Europe/Berlin:20150517T090000
END:VEVENT
```
The Event had been somehow overwritten. The `UID` stays the same but the event includes an attribute `RECURRENCE-ID`.

This is what [rfc 4791](https://tools.ietf.org/html/rfc4791#section-4.1) says:
>  Calendar components with the same UID property value, in a given calendar collection, MUST be contained in the same calendar object resource.  This ensures that all components in a recurrence "set" are contained in the same calendar object resource.  It is possible for a calendar object resource to just contain components that represent "overridden" instances (ones that modify the behavior of a regular instance, and thus include a RECURRENCE-ID property) without also including the "master" recurring component (the one that defines the recurrence "set" and does not contain any RECURRENCE-ID property).

**Comment by jhnns** [comment]
Note to future flootr!

**Comment by flootr** [comment]
rfc 4791 3.2
> Clients are not forced to retrieve information about all recurrence instances of a recurring component.  The CALDAV:calendar-query and CALDAV:calendar-multiget reports defined in this document allow clients to retrieve only recurrence instances that overlap a given time range.

**Comment by antosan** [comment]
Fixed in v0.2.0
